crypto/rand_test.go

```diff
@@ -4,13 +4,26 @@ import (
 	"testing"
 )
 
+// panics returns true if the function fn panicked.
+func panics(fn func()) (panicked bool) {
+	defer func() {
+		panicked = (recover() != nil)
+	}()
+	fn()
+	return
+}
+
 // TestRandIntnPanics tests that RandIntn panics if n <= 0.
 func TestRandIntnPanics(t *testing.T) {
-	defer func() {
-		if r := recover(); r == nil {
+	testNs := []int{0, -1}
+	for _, n := range testNs {
+		if !panics(func() {
+			_, err := RandIntn(n)
+			if err != nil {
+				t.Error("expected panic on n <= 0, not error")
```

Comment:
this error is kinda redundant: if it's triggered, then the `t.Error` below will be triggered as well.